File: ai-job-portal-backend/candidate-service/src/main/java/com/prahlad/aijobportal/candidateservice/skill/service/SkillService.java
package com.prahlad.aijobportal.candidateservice.skill.service;

import com.prahlad.aijobportal.candidateservice.skill.dto.request.SkillRequest;
import com.prahlad.aijobportal.candidateservice.skill.dto.response.SkillResponse;

import java.util.List;
import java.util.UUID;

public interface SkillService {

    SkillResponse create(UUID userId, SkillRequest request);

    List<SkillResponse> getAll(UUID userId);

    SkillResponse update(UUID userId, UUID skillId, SkillRequest request);

    void delete(UUID userId, UUID skillId);
}

File: ai-job-portal-frontend/src/features/profile/services/skill.service.ts
import { apiClient } from "@/services/api-client";
import type { ApiResponse } from "@/types/api";
import type { SkillRequest, SkillResponse } from "@/features/profile/types";

/** Maps 1:1 to candidate-service SkillController (/candidate/skills). */
export const skillService = {
  getAll: () => apiClient.get<ApiResponse<SkillResponse[]>>("/candidate/skills").then((res) => res.data),

  create: (payload: SkillRequest) =>
    apiClient.post<ApiResponse<SkillResponse>>("/candidate/skills", payload).then((res) => res.data),

  update: (skillId: string, payload: SkillRequest) =>
    apiClient.put<ApiResponse<SkillResponse>>(`/candidate/skills/${skillId}`, payload).then((res) => res.data),

  delete: (skillId: string) =>
    apiClient.delete<ApiResponse<null>>(`/candidate/skills/${skillId}`).then((res) => res.data),
};
